backend/app/services/langgraph_chat/tools/mcp/client.py

```
@@ -0,0 +1,408 @@
+    async def _send_request(self, method: str, params: Dict[str, Any]) -> Dict[str, Any]:
+        """Send JSON-RPC request to MCP server.
+
+        Args:
+            method: RPC method name
+            params: Method parameters
+
+        Returns:
+            Response data
+        """
+        if not self.process or not self.process.stdin or not self.process.stdout:
+            raise RuntimeError("Process not running")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)